### PR TITLE
feat: DashPay | Explore | Where to Spend

### PR DESCRIPTION
**Pull request by tikhop**
## Issue being fixed or feature implemented
- [x] Implement explore welcome screen
- [x] Implement where to spend info screens
- [x] Implement where to spend screen using mock data  

## How Has This Been Tested?
On Device and Simulator

## Breaking Changes
N/A

## Checklist:

- [x] I have performed a self-review of my own code
- [x] I have commented my code, particularly in hard-to-understand areas
- [x] I have added or updated relevant unit/integration/functional/e2e tests
- [x] I have made corresponding changes to the documentation

**For repository code-owners and collaborators only**
- [ ] I have assigned this pull request to a milestone

### CODE REVIEW

File: DashWallet/Sources/UI/DashPay/Explore/Views/DWExploreHeaderView.m [hunk 1]
```diff
@@ -33,15 +35,20 @@ @interface DWExploreHeaderView ()
 
 @implementation DWExploreHeaderView
 
+CGFloat const kExploreHeaderViewHeight = 351.0f;
```
Comment (QuantumExplorer) [inline DashWallet/Sources/UI/DashPay/Explore/Views/DWExploreHeaderView.m:38]: What is this for? Not great hardcoding things like this.

File: DashWallet/Sources/UI/DashPay/Explore/WhereToSpend/Cells/DWExploreWhereToSpendFiltersCell.m [hunk 2]
```diff
@@ -0,0 +1,94 @@
...
+    [filterButton addTarget:self
+                    action:@selector(filterButtonAction)
+          forControlEvents:UIControlEventTouchUpInside];
+    [stackView addArrangedSubview:filterButton];
+    _filterButton = filterButton;
+    
+    [NSLayoutConstraint activateConstraints:@[
+        [filterButton.widthAnchor constraintEqualToConstant:90],
+        [stackView.centerYAnchor constraintEqualToAnchor:self.contentView.centerYAnchor],
+        //[segmentedControl.bottomAnchor constraintEqualToAnchor:self.contentView.bottomAnchor],
```
Comment (QuantumExplorer) [inline DashWallet/Sources/UI/DashPay/Explore/WhereToSpend/Cells/DWExploreWhereToSpendFiltersCell.m:88]: Why commented out?

File: DashWallet/Sources/UI/DashPay/Explore/WhereToSpend/Cells/DWExploreWhereToSpendSearchCell.m [hunk 3]
```diff
@@ -0,0 +1,56 @@
...
+
+-(void)configureHierarchy {
+    self.searchBar = [[UISearchBar alloc] init];
+    _searchBar.translatesAutoresizingMaskIntoConstraints = NO;
+    _searchBar.searchBarStyle = UISearchBarStyleMinimal;
+    [self.contentView addSubview:_searchBar];
+    
+    [NSLayoutConstraint activateConstraints:@[
+        [_searchBar.centerYAnchor constraintEqualToAnchor:self.contentView.centerYAnchor],
+        //[segmentedControl.bottomAnchor constraintEqualToAnchor:self.contentView.bottomAnchor],
```
Comment (QuantumExplorer) [inline DashWallet/Sources/UI/DashPay/Explore/WhereToSpend/Cells/DWExploreWhereToSpendSearchCell.m:50]: Why commented out?

File: DashWallet/Sources/UI/DashPay/Explore/WhereToSpend/DWExploreGiftCardInfoViewController.m [hunk 4]
```diff
@@ -0,0 +1,120 @@
...
+@implementation DWExploreGiftCardInfoViewController
+
+-(void)configureHierarchy {
+    [super configureHierarchy];
+    
+    UIStackView *contentView = [UIStackView new];
+    contentView.axis = UILayoutConstraintAxisVertical;
+    contentView.spacing = 30;
+    contentView.translatesAutoresizingMaskIntoConstraints = NO;
+    //contentView.distribution = UIStackViewDistributionEqualSpacing;
```
Comment (QuantumExplorer) [inline DashWallet/Sources/UI/DashPay/Explore/WhereToSpend/DWExploreGiftCardInfoViewController.m:34]: Why commented out?

File: DashWallet/Sources/UI/DashPay/Explore/WhereToSpend/DWExploreGiftCardInfoViewController.m [hunk 5]
```diff
@@ -0,0 +1,120 @@
...
+        
+    ]];
+}
+
+-(UIView *)itemViewFor:(NSString *)title image:(UIImage *)image {
+    UIStackView *itemStackView = [UIStackView new];
+    itemStackView.axis = UILayoutConstraintAxisHorizontal;
+    itemStackView.spacing = 10;
+    itemStackView.translatesAutoresizingMaskIntoConstraints = NO;
+    //itemStackView.distribution = UIStackViewDistributionFillProportionally;
```
Comment (QuantumExplorer) [inline DashWallet/Sources/UI/DashPay/Explore/WhereToSpend/DWExploreGiftCardInfoViewController.m:89]: why commented out?

File: DashWallet/Sources/UI/DashPay/Explore/WhereToSpend/DWExploreGiftCardInfoViewController.m [hunk 6]
```diff
@@ -0,0 +1,120 @@
...
+    itemTitleLabel.font = [UIFont dw_fontForTextStyle:UIFontTextStyleBody];
+    itemTitleLabel.textAlignment = NSTextAlignmentLeft;
+    itemTitleLabel.numberOfLines = 0;
+    [itemStackView addArrangedSubview:itemTitleLabel];
+    
+    [NSLayoutConstraint activateConstraints:@[
+        [iconImageView.widthAnchor constraintEqualToConstant:50],
+        
+        [iconImageView.centerYAnchor constraintEqualToAnchor:itemTitleLabel.topAnchor constant:10],
+        //[itemTitleLabel.topAnchor constraintEqualToAnchor:iconImageView.centerYAnchor constant:5],
```
Comment (QuantumExplorer) [inline DashWallet/Sources/UI/DashPay/Explore/WhereToSpend/DWExploreGiftCardInfoViewController.m:110]: commented out

**Review by QuantumExplorer — APPROVED** [review]

**Review by pankcuf — APPROVED** [review]
probably, we'll need to use another commit version...

**Pull request closed without merging**